src/software/test_util/test_util.cpp

```diff
@@ -104,35 +106,84 @@ namespace Test
     }
 
     bool TestUtil::equalWithinTolerance(const Polygon &poly1, const Polygon &poly2,
-                                        double tolerance)
+                                        double tolerance, bool debug)
     {
         auto ppts1 = poly1.getPoints();
         auto ppts2 = poly2.getPoints();
-        return std::equal(ppts1.begin(), ppts1.end(), ppts2.begin(),
-                          [tolerance](const Point &p1, const Point &p2) {
-                              return equalWithinTolerance(p1, p2, tolerance);
-                          });
+        if (std::equal(ppts1.begin(), ppts1.end(), ppts2.begin(),
+                       [tolerance](const Point &p1, const Point &p2) {
```

Comment:
Does this only work of the Points that make up the polygon are in the same order?

Reply:
yes